File: src/0941_valid_mountain_array/valid_mountain_array.py
# coding:utf8

from typing import *

class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3 :
            return False

        i = 1
        flagIncrease = False # 上升
        flagDecrease = False # 下降

        while i < len(arr) and arr[i-1] < arr[i]:
            flagIncrease = True
            i += 1

        while i < len(arr) and arr[i-1] > arr[i]:
            flagDecrease = True
            i += 1

        return i == len(arr) and flagIncrease and flagDecrease

File: src/0941_valid_mountain_array/valid_mountain_array.go
// author: yqq
// date: 2021-09-19 18:21:31
// descriptions:
package main
import "fmt"

func validMountainArray(arr []int) bool {
	if len(arr) < 3 {
		return false
	}

	i := 1
	flagIncrease := false // 上升
	flagDecrease := false // 下降
	
	for ; i < len(arr) && arr[i-1] < arr[i]; i++ {
		flagIncrease = true;
	}

	for ; i < len(arr) && arr[i-1] > arr[i]; i++ {
		flagDecrease = true;
	}

	return i == len(arr) && flagIncrease && flagDecrease;
}


func main() {
    fmt.Println("hello world")
}

